libs/media/api/src/lib/media-service/transform.guard.ts

```diff
@@ -47,17 +50,23 @@ export class TransformGuard {
     return metadata.pages && metadata.pages > 1
   }
 
+  public validateSignature(imageId: string, t: TransformationsDto) {
+    const {sig, ...dataWithoutSignature} = t
+    const validationSignature = getSignatureForImage(imageId, dataWithoutSignature)
+
+    if (!timeConstantCompare(sig ?? '', validationSignature)) {
+      throw new ForbiddenException('Invalid signature!')
+    }
+    return true
+  }
+
   public checkDimensions(metadata: sharp.Metadata, transformations: TransformationsDto): number {
-    const originalHeight = metadata.height
```

Comment:
So: Could an uploader fake the height of an image?

Reply:
What do you mean by fake?